sourced/ml/utils/spark.py

```diff
@@ -29,7 +30,7 @@ def add_spark_args(my_parser, default_packages=None):
         "--config", nargs="+", default=SparkDefault.CONFIG,
         help="Spark configuration (key=value).")
     my_parser.add_argument(
-        "-m", "--memory",
+        "-m", "--memory", default=SparkDefault.MEMORY,
```

Comment:
This is a bad default, it explodes on any big query. `DISK` + `MEMORY` is better

Reply:
done

Reply:
I think you are confusing the `--persist` flag that specifies storage level when caching, and this `--memory` flag that is just a shortcut to define  executor memory, driver memory and driver max result size, and currently defaults to nothing. but I agress we should have persistence set to `MEMORY` and `DISK`, changing

Reply:
Right, I confused them